CMakeLists.txt

```diff
@@ -27,6 +32,31 @@ set_target_properties(segyio-static PROPERTIES COMPILE_FLAGS "-fPIC")
 add_library(segyio-shared SHARED ${SOURCE_FILES})
 set_target_properties(segyio-shared PROPERTIES OUTPUT_NAME segyio CLEAN_DIRECT_OUTPUT 1)
 
+
+if(BUILD_PYTHON)
+    find_package(PythonInterp)
+    find_package(PythonLibs REQUIRED)
+
+    if (PYTHONLIBS_FOUND)
+        include_directories(${PYTHON_INCLUDE_DIRS})
+        add_library(_segyio SHARED python/segyio/_segyio.c ${SOURCE_FILES})
+
+        if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            set_target_properties(_segyio PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+        else()
+            set_target_properties(_segyio PROPERTIES LINK_FLAGS "-Xlinker -export-dynamic")
+        endif()
+
+        set_target_properties(_segyio PROPERTIES PREFIX "" SUFFIX ".so")
+        install(TARGETS _segyio DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/python2.7/site-packages/segyio)
```

Comment:
I have incorporated these fixes into PR #16 which include additional refactoring and improvements to CMake and Travis.